cmd/infra/powervs/destroy.go

```diff
@@ -206,6 +222,50 @@ func (options *DestroyInfraOptions) DestroyInfra(infra *Infra) (err error) {
 	return
 }
 
+// deleteDNSRecords deletes DNS records from CIS domain
+func deleteDNSRecords(options *DestroyInfraOptions) error {
+
+	if options.CisCRN == "" || options.CisDomainID == "" {
+		if options.BaseDomain == "" || options.Name == "" {
+			log(options.InfraID).Info("Skipping DNS record deletion. DNS Record deletion requires --name & --base-domain")
+			return nil
+		} else {
+			var err error
+			options.CisCRN, options.CisDomainID, err = getCisDomainDetails(options.BaseDomain)
+			if err != nil {
+				return fmt.Errorf("error retrieving cis domain details %w", err)
+			}
+		}
+	}
+
+	dnsRecordsV1, err := dnsrecordsv1.NewDnsRecordsV1(&dnsrecordsv1.DnsRecordsV1Options{Crn: &options.CisCRN, ZoneIdentifier: &options.CisDomainID, Authenticator: getIAMAuth()})
```

Comment:
There is a possibility of `options.CisCRN, options.CisDomainID` can be empty string, if they are empty then do we really need to proceed further?

case: where user deletes the dns entries manually

Reply:
No we are depending the base-domain's CRN and DomainID only right, if they remove DNS entry, we can still get CRN and DomainID info. 
If they remove base-domain itself, it will be empty, in that case `getCisDomainDetails` will return an error and we won't proceed.